### PR TITLE
Add ENV var for metadata download location

### PR DESCRIPTION
**Pull request by zachthompson**
* Add and environment variable - `METADATA_TMP_DIR` -` to specify different metadata tmp directories
* Default to `/var/tmp`

cc @jkv

### CODE REVIEW

File: lib/DDG/Meta/Data.pm [hunk 1]
```diff
@@ -24,12 +24,9 @@ my %ia_metadata;
...
 unless(%ia_metadata){
 
-    my $tmpdir = io->tmpdir;
-    unless($tmpdir){
-       die 'No system temp directory found';
-    }
+    my $tmpdir = $ENV{METADATA_TMP_DIR} || '/var/tmp';
 
-    my $mdir = "$tmpdir/ddg-$>";
+	my $mdir = "$tmpdir/ddg-$>";
```
Comment (malbin) [inline lib/DDG/Meta/Data.pm]: intentionally replacing spaces with tab?

File: lib/DDG/Meta/Data.pm [hunk 2]
```diff
@@ -24,12 +24,9 @@ my %ia_metadata;
 # Only build metadata once. Not in BUILD so we can call apply_keywords directly
 unless(%ia_metadata){
 
-    my $tmpdir = io->tmpdir;
-    unless($tmpdir){
-       die 'No system temp directory found';
-    }
+    my $tmpdir = $ENV{METADATA_TMP_DIR} || '/var/tmp';
```
Comment (malbin) [inline lib/DDG/Meta/Data.pm]: could we default to `/var/tmp/ddg-metadata/` as per @jkv's suggestion from the Asana desc?

File: lib/DDG/Meta/Data.pm [hunk 3]
```diff
@@ -24,14 +25,11 @@ my %ia_metadata;
...
-    unless($tmpdir){
-       die 'No system temp directory found';
-    }
+    my $tmpdir = $ENV{METADATA_TMP_DIR} || '/var/tmp/ddg-metadata';
 
-    my $mdir = "$tmpdir/ddg-$>";
+    my $mdir = "$tmpdir/$>";
     unless(-d $mdir){
-        mkdir $mdir or die "Failed to mkdir $mdir: $!";
+        pathmk $mdir or die "Failed to mkdir $mdir: $!";
```
Comment (malbin) [inline lib/DDG/Meta/Data.pm]: Is this recursive? (will it emulate `mkdir -p` ?)

Comment (zachthompson) [inline lib/DDG/Meta/Data.pm]: yep

**Comment by zachthompson** [comment]
Yo @jdorweiler, what say you?

**Comment by jdorweiler** [comment]
@zachthompson  looks good.  I tested it and works fine.  In what cases would you want to set the env variable?

**Comment by jdorweiler** [comment]
@zachthompson looks good for real this time. :cake:

**Comment by zachthompson** [comment]
@jdorweiler A user might need to specify an alternate location if they, say, don't have permissions to /var/tmp.  Or maybe we need to switch locations temporarily if /var/tmp were on a mount that went offline.  Those types of things.

**Pull request merged**